kqueen/blueprints/api/generic_views.py

```diff
@@ -192,7 +192,29 @@ def _save_objects_range(self, objects):
         else:
             self.obj = sorted_by_date
 
+    def _filter_objects(self, objects, filters):
+        def get_name(obj, attr_name):
+            name = getattr(obj, attr_name, '')
+            if attr_name == 'name':
+                return name
+            return getattr(name, 'name', '')
+
+        if not filters:
+            return objects
+        return list(filter(
+            lambda x: all(filters[key].lower() in get_name(x, key).lower()
+                          for key in filters.keys()),
+            objects
+        ))
+
     def set_object(self, *args, **kwargs):
+        supported_filters = ['name', 'provisioner']
```

Comment:
IMO it will be better to rename 'name' to `cluster_name`

Reply:
It's not only for clusters